project/Build.scala

```diff
@@ -832,6 +832,13 @@ object Build {
       },
       scalacOptions += "-scalajs",
 
+      // Replace the JVM JUnit dependency by the Scala.js one
+      libraryDependencies ~= {
+        _.filter(!_.name.startsWith("junit-interface"))
+      },
+      libraryDependencies +=
+        "org.scala-js" % "scalajs-junit-test-runtime_2.12" % scalaJSVersion  % "test",
```

Comment:
```suggestion
        ("org.scala-js" %% "scalajs-junit-test-runtime" % scalaJSVersion  % "test").withDottyCompat(scalaVersion.value),
```

Also is this something normally done by the sbt plugin for Scala.js ?

Reply:
The *addition* is normally done by the `ScalaJSJUnitPlugin`, which is separate from the base `ScalaJSPlugin`. That plugin also enables the *compiler plugin* for the additional phase in scalac, btw.

The removal of the JVM one is not done in any of those, because you're not supposed to add it in the first place. But in this build the `commonSettings` put the JUnit dependency on everything, and I didn't feel like interfering with the rest of the build at the moment.